src/r5py/util/memory_footprint.py

```diff
@@ -60,29 +60,64 @@ def share_of_ram(share=0.8, leave_at_least=(2 * (2**10))):
     return share_of_ram
 
 
-def max_memory(max_memory):
-    """Interpret the config parameter --max-memory."""
+def parse_max_memory_string(max_memory):
+    """
+    Extract maximum memory value and unit from text input.
+    """
     try:
-        matches = re.match(r"(?P<value>[0-9]+(\.[0-9]+)?)(?P<unit>[%MGT])?", max_memory)
+        matches = re.match(
+            r"(?P<value>[0-9]+(\.[0-9]+)?)(?P<unit>[%KMGT])?", max_memory
+        )
         value = float(matches["value"])
         unit = matches["unit"]
-        if unit == "%":
-            max_memory = share_of_ram(share=(value / 100.0))
-        else:
-            # convert to MiB
-            if unit is None:
-                value *= 2**-10
-                if value < 1:
-                    value = 1
-            # elif unit == "M":
-            #    value *= 2 ** 1
-            elif unit == "G":
-                value *= 2**10
-            elif unit == "T":
-                value *= 2**20
-            max_memory = round(value)
+
+        if unit is None:
+            raise ValueError(
+                "Could not interpret the memory unit from --max-memory."
+                "The suffix for --max-memory should be '%', 'K', 'M', 'G' or 'T'."
+                "For example to allocate five gigabytes of memory, use: '5G'"
+            )
+        return value, unit
     except TypeError:
-        raise ValueError(f"Could not interpret --max-memory: {max_memory}")
+        raise ValueError(
+            f"Could not interpret --max-memory: {max_memory}."
+            f"To allocate memory, use e.g. '5G' for five gigabytes of memory."
+        )
+
+
+def get_max_memory(max_memory):
+    """
+    Interpret the config parameter --max-memory.
+
+    Arguments
+    ---------
+
+    max_memory : str
+        Memory limit for the JVM running R5.
+
+        Use % as a suffix to specify a share of total RAM;
+        K, M, G, T suffix specify KiB, MiB, GiB, or TiB, respectively.
+        Values without suffix are interpreted as bytes.
+        Values are rounded to the closest MiB.
+    """
+
+    value, unit = parse_max_memory_string(max_memory)
+
+    if unit == "%":
+        max_memory = share_of_ram(share=(value / 100.0))
+    else:
+        # convert to MiB
+        if unit == "K":
+            value *= 2**-20
+            if value < 1:
+                value = 1
```

Comment:
I think this logic (`value = max(value, 1)`) should be moved to after the `round()`ing.

Reply:
Done.